### PR TITLE
[MRG+1] DOC: Add bug report reqs and template to contributing guide

### PR DESCRIPTION
**Pull request by petehuang**
Hopefully this doesn't conflict with #7699 at all (it shouldn't).

This adds a few recommendations for information to include when submitting bug reports. It also provides a Markdown-formatted template for anyone who wants to neatly organize their information in their issue.

### CODE REVIEW

**Comment by NelleV** [comment]
I'm 50%-50% on this patch: it duplicates a lot of elements which we already have on our ticket page: https://github.com/matplotlib/matplotlib/issues/new

Could we have all of this in a single place?

In terms of content, it looks good to me.

**Comment by petehuang** [comment]
Oops, wasn't aware of the pre-loaded template on the new issue page. Let's definitely sync from both ends. Is there any place that template can be changed?

**Comment by NelleV** [comment]
You can find it here `.github/ISSUE_TEMPLATE.md`
I think it needs a bit of love :)

File: doc/devel/contributing.rst [hunk 1]
```diff
@@ -14,8 +14,60 @@ Submitting a bug report
...
+
+ 2. A short, self-contained code snippet to reproduce the bug, ideally allowing
+    a simple copy and paste to reproduce. Please do your best to reduce the code 
+    snippet to the minimum required.
+
+ 3. The actual outcome of the code snippet
+
+ 4. The expected outcome of the code snippet
+
+ 5. The matplotlib version, python version and platform that you are using. You can grab the
```
Comment (NelleV) [inline doc/devel/contributing.rst]: @efiring @tacaswell So should matplotlib be capitalized here?

File: doc/devel/contributing.rst [hunk 2]
```diff
@@ -14,8 +14,60 @@ Submitting a bug report
...
+        '1.5.3'
+        >>> import platform
+        >>> platform.python_version()
+        '2.7.12'
+
+We have preloaded the issue creation page with a Markdown template that you can use to organize this information.
+        
+Thank you for your help in keeping bug reports complete, targeted and descriptive.
+
+Submitting a feature request
```
Comment (NelleV) [inline doc/devel/contributing.rst]: I'd remove this section. I don't think it is very useful and has a lot of duplication with the previous section. I think we need to keep the contributing guideline short.

File: doc/devel/contributing.rst [hunk 3]
```diff
@@ -14,8 +14,60 @@ Submitting a bug report
 
 If you find a bug in the code or documentation, do not hesitate to submit a
 ticket to the
-`Bug Tracker <https://github.com/matplotlib/matplotlib/issues>`_. You are
-also welcome to post feature requests or pull requests.
+`Bug Tracker <https://github.com/matplotlib/matplotlib/issues>`_.
```
Comment (NelleV) [inline doc/devel/contributing.rst]: I would add back the sentence "You are also welcome to post feature requests or pull requests" here, and remove the section down of details about the feature requests.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -1,11 +1,52 @@
...
+```
+# If applicable, paste the console output here
+#
+#
+```
+
+**Expected outcome**
+
+- [ ] A description of the expected outcome from the code snippet
+- [ ] If this used to work in an earlier version of Matplotlib, please note where it used to work
```
Comment (NelleV) [inline .github/ISSUE_TEMPLATE.md]: "please note the version it used to work on?"

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -1,11 +1,52 @@
...
 - [ ] Matplotlib version, Python version and Platform (Windows, OSX, Linux ...)
 - [ ] How did you install Matplotlib and Python (pip, anaconda, from source ...)
-- [ ] If possible please supply a [Short, Self Contained, Correct, Example](http://sscce.org/)
-      that demonstrates the issue i.e a small piece of code which reproduces the issue
-      and can be run with out any other (or as few as possible) external dependencies.
-- [ ] If this is an image generation bug attach a screenshot demonstrating the issue.
-- [ ] If this is a regression (Used to work in an earlier version of Matplotlib), please 
-      note where it used to work.
+
+###Feature request
```
Comment (NelleV) [inline .github/ISSUE_TEMPLATE.md]: I'd remove this section. We want to keep this as simple and short as possible.

**Comment by efiring** [comment]
On 2016/12/30 9:29 PM, Nelle Varoquaux wrote:
> + 5. The matplotlib version, python version and platform that you are using. You can grab the
>
> @efiring <https://github.com/efiring> @tacaswell
> <https://github.com/tacaswell> So should matplotlib be capitalized here?
>

I would say "yes", but this is one of those places where it is perfectly 
acceptable to me either way.

File: doc/devel/contributing.rst [hunk 6]
```diff
@@ -14,9 +14,34 @@ Submitting a bug report
 
 If you find a bug in the code or documentation, do not hesitate to submit a
 ticket to the
-`Bug Tracker <https://github.com/matplotlib/matplotlib/issues>`_. You are
-also welcome to post feature requests or pull requests.
+`Bug Tracker <https://github.com/matplotlib/matplotlib/issues>`_. You are also welcome to post feature requests or pull requests.
```
Comment (QuLogic) [inline doc/devel/contributing.rst]: Wrap at 80 here too.

**Comment by NelleV** [comment]
LGTM :+1:

**Pull request merged**